src/transformNestedSelection.ts

```diff
@@ -106,6 +109,16 @@ const unprefixTypes = (document: DocumentNode, prefix: string) => {
   }) as DocumentNode
 }
 
+// If the nested schema was extended, those types can appear in the selection set but should not be sent to the endpoint
+const pruneExtendedFields = (schema: GraphQLSchema, document: DocumentNode): DocumentNode => {
+  const typeInfo = new TypeInfo(schema)
+  return visit(document, visitWithTypeInfo(typeInfo, {
+    Field(node) {
+      return typeInfo.getFieldDef() ? node : null
```

Comment:
+1 returning `undefined` instead of `node` can save a few cycles

https://github.com/graphql/graphql-js/blob/main/src/language/visitor.ts#L267-L276